### PR TITLE
Rest API

### LINKED ISSUE #223: Fix /get/{policy}/{volume} REST API
It currently returns
```
{
  "Volume": {
    "Name": "policy1/postgres-vol",
    "Mountpoint": "/mnt/ceph/rbd/policy1.postgres-vol"
  },
  "Err": ""
}
```
Expect it to return same output as `volcli volume get policy1/postgres-vol`:
```
{
  "policy": "policy1",
  "name": "postgres-vol",
  "driver": {
    "pool": "rbd"
  },
  "create": {
    "size": "10MB",
    "filesystem": "ext4"
  },
  "runtime": {
    "snapshots": true,
    "snapshot": {
      "frequency": "30m",
      "keep": 20
    },
    "rate-limit": {
      "write-iops": 0,
      "read-iops": 0,
      "write-bps": 0,
      "read-bps": 0
    }
  },
  "backend": "ceph"
}
```

### LINKED ISSUE #226: Add REST endpoints for policy CRUD
REST endpoints to do CRUD of storage policies.
volcli equivalent of:
```
[vagrant@mon0 ~]$ volcli policy list
policy1
[vagrant@mon0 ~]$ volcli policy get policy1
{
  "create": {
    "size": "10MB",
    "filesystem": ""
  },
  "runtime": {
    "snapshots": true,
    "snapshot": {
      "frequency": "30m",
      "keep": 20
    },
    "rate-limit": {
      "write-iops": 0,
      "read-iops": 0,
      "write-bps": 0,
      "read-bps": 0
    }
  },
  "driver": {
    "pool": "rbd"
  },
  "filesystems": {
    "ext4": "mkfs.ext4 -m0 %"
  },
  "backend": "ceph"
}
```

Also for upload and delete.

### PR DESCRIPTION
**Pull request by erikh**
Fixes #226 
Fixes #223 

/cc @vishal-j

### CODE REVIEW

File: volmaster/daemon.go [hunk 1]
```diff
@@ -94,6 +82,7 @@ func (d *DaemonConfig) Daemon(debug bool, listen string) {
 	}
 
 	getRouter := map[string]func(http.ResponseWriter, *http.Request){
+		"/policy/{policy}":             d.handlePolicy,
```
Comment (mapuri) [inline volmaster/daemon.go]: not related to this PR but looks like we have slight inconsistency between different GET endpoints, like for policy we have `policy/{policy}` but for volumes we have `/get/{policy}/{volume}` may be volume should also be accessible as `volume/{policy}/{volume}` ?

File: volmaster/daemon.go [hunk 2]
```diff
@@ -137,6 +126,25 @@ func (d *DaemonConfig) handleDebug(w http.ResponseWriter, r *http.Request) {
...
+
+	policyObj, err := d.Config.GetPolicy(policy)
+	if err != nil {
+		httpError(w, "Retrieving policy", err)
+		return
+	}
+
+	content, err := json.Marshal(policyObj)
+	if err != nil {
+		httpError(w, "Marshalling policy repsonse", err)
```
Comment (mapuri) [inline volmaster/daemon.go]: `repsonse` -> `response`

File: volplugin/handlers.go [hunk 3]
```diff
@@ -114,7 +114,51 @@ func (dc *DaemonConfig) get(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	path, err := driver.MountPath(do)
+	if err != nil {
+		httpError(w, "Calculating mount path", err)
+		return
+	}
+
+	fmt.Println(path)
```
Comment (mapuri) [inline volplugin/handlers.go]: this looks like a left over print or should be `log.Debugf()`?

File: volplugin/handlers.go [hunk 4]
```diff
@@ -141,6 +228,8 @@ func (dc *DaemonConfig) getPath(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	fmt.Println(uc)
```
Comment (mapuri) [inline volplugin/handlers.go]: this looks like a left over print or should be `log.Debugf()`?

**Comment by mapuri** [comment]
LGTM. just few minor nits.

File: volmaster/daemon.go (hunk 1 above)
Comment (erikh) [inline volmaster/daemon.go]: Let's put in an issue for this one. The whole hierarchy probably needs a  touch of rework.

**Comment by erikh** [comment]
This is ready for another pass.

**Comment by mapuri** [comment]
LGTM

**Pull request merged**